demos/echo.rst

```diff
@@ -0,0 +1,93 @@
+
+.. _echo-test-remote-app:
+
+RPmsg Echo Remote Application
+=============================
+
+The remote application, rpmsg-echo, is the core of the demonstration. It is a simple application serving a :ref:`RPmsg endpoint <rpmsg-endpoint>` running as the main task on the remote processor, once loaded and started using :ref:`Remoteproc<overview-remoteproc-work-label>`.
+
+
+.. _echo-test-host-app:
+
+Echo Test Host Application
+==========================
+
+The echo_test application forms the host controller side of the demonstration. It repeatedly writes an increasing length payload of 0xA5's up to the maximum data size (packet size minus header) to the RPmsg endpoint. Following each packet send, it reads from the same endpoint and verifies the returned packet for correctness. The application will stop and report on the first corruption found.
+
+Echo Test Host Script
+=====================
+
+The host is also responsible for loading the firmware containing the :ref:`RPmsg Echo Remote Application<echo-test-remote-app>` and starting the remote processor using :ref:`Remoteproc<overview-remoteproc-work-label>`.
+
+For host controllers, like Linux, a script can be used to pipe the firmware to the exposed remoteproc system, followed by the execution of the user space echo_test application. For controllers without scripting capability, like baremetal and RTOS (Real Time Operating systems), this would be achieved in the code.
+
+****************
+Echo Test Source
+****************
+
+RPmsg Echo Baremetal Source
+===========================
+
+The RPmsg Echo service application is available as a baremetal solution in the `open-amp Repository <https://github.com/OpenAMP/open-amp/blob/main/apps/examples/echo/rpmsg-echo.c>`_
+
+It is a CMake application and can be built for any remote as long as the relevant :ref:`OS/HW abstraction layer<porting-guide-work-label>` components like libmetal are ported for that platform.
+
+.. _echo-test-linux-app:
+
+Echo Test Linux Source
+======================
+
+The echo test Linux application is executed on the Linux host controller as a user space application.
+The application is available in the `OpenAMP System Reference repository <https://github.com/OpenAMP/openamp-system-reference/blob/main/examples/linux/rpmsg-echo-test/echo_test.c>`_.
+
+It is a Makefile application and can be built using the `Yocto rpmsg-echo-test recipe <https://github.com/OpenAMP/meta-openamp/blob/master/recipes-openamp/rpmsg-examples/rpmsg-echo-test_1.0.bb>`_
+
+An example host control script is given in the `echo test readme <https://github.com/OpenAMP/openamp-system-reference/blob/main/examples/linux/rpmsg-echo-test/README.md#run-the-demo>`_
+
+*******************************
+Reference Board Implementations
+*******************************
+
+This Echo Test Sample is demonstrated in the following reference implementations.
+
+* :ref:`Docker Images<docker-images-label>` as demo1A
+* :ref:`AMD-Xilinx platforms<demos-AMD--work-label>`
```

Comment:
It is also possible to run this test on a linux PC between 2 processes as described here:
https://github.com/OpenAMP/open-amp/blob/main/README.md#example-to-compile-openamp-for-communication-between-linux-processes

Reply:
added a todo for later addition

Reply:
done